astropy/coordinates/angles.py

```diff
@@ -668,23 +667,20 @@ def __array_wrap__(self, obj, context=None):
         return obj
 
 #<----------------------------------Rotations--------------------------------->
-
-
 def rotation_matrix(angle, axis='z', unit=None):
     """
-    Generate a 3x3 cartesian rotation matrix in for rotation about
-    a particular axis.
+    Generate rotation matrices for rotation by some angle(s).
 
     Parameters
     ----------
     angle : convertible to `Angle`
-        The amount of rotation this matrix should represent.
+        The amount of rotation the matrices should represent.
 
-    axis : str or 3-sequence
-        Either ``'x'``, ``'y'``, ``'z'``, or a (x,y,z) specifying an
-        axis to rotate about. If ``'x'``, ``'y'``, or ``'z'``, the
-        rotation sense is counterclockwise looking down the + axis
-        (e.g. positive rotations obey left-hand-rule).
+    axis : str, or array-like
+        Either ``'x'``, ``'y'``, ``'z'``, or a (x,y,z) specifying the axis to
+        rotate about. If ``'x'``, ``'y'``, or ``'z'``, the rotation sense is
+        counterclockwise looking down the + axis (e.g. positive rotations obey
+        left-hand-rule).  If given as an array, the last dimension should be 3.
```

Comment:
'x', 'y', and 'z' are mostly meant as short-cuts to speed up those common cases; the speed-up would certainly be lost for multiple values, and since this is used mostly internally anyway, I'd tend to leave it alone.